lib/segment/src/types.rs

```diff
@@ -78,6 +76,31 @@ impl FromStr for ExtendedPointId {
     }
 }
 
+impl<'de> serde::Deserialize<'de> for ExtendedPointId {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let value = match serde_value::Value::deserialize(deserializer) {
+            Ok(val) => val,
+            Err(err) => return Err(err),
+        };
+
+        if let Ok(num) = value.clone().deserialize_into() {
+            return Ok(ExtendedPointId::NumId(num));
+        }
+
+        if let Ok(uuid) = value.clone().deserialize_into() {
+            return Ok(ExtendedPointId::Uuid(uuid));
+        }
+
+        Err(serde::de::Error::custom(format!(
+            "value `{}` did not match any variant of untagged enum ExtendedPointId",
```

Comment:
Do you have maybe an example for the error message to see how it looks like concretely for the users?

Reply:
Well, it would (partially) depend on what kind of value was in the input.

- if `serde_value::Value` deserialized into one of basic type variants
  - (which, in case of JSON, means the input was a simple value, like `10` or `"string"` or `true`)
  - then it would simply `Display` the value of this variant, e.g.:
  - ```value `random string` did not match any variant of untagged enum ExtendedPointId```
  - or ```value `0.0000042` did not match any variant of untagged enum ExtendedPointId```
- if `serde_value::Value` deserialized into one of the other variants, then it would `Debug` the value, e.g.:
  - ```value `Map(BTreeMap { ... })` did not match any variant of untagged enum ExtendedPointId```

I've created the PR, mostly to see CI run results, so we can discuss/adjust the message to something more user-friendly.

Reply:
Reworded a bit, to match the message in that commit that was reverted after release.

Reply:
Thanks for the details, I am mostly interested in the case string and numeric :+1: